base/compiler/tfuncs.jl

```diff
@@ -518,13 +522,19 @@ function subtype_tfunc(@nospecialize(a), @nospecialize(b))
 end
 add_tfunc(<:, 2, 2, subtype_tfunc, 0)
 
+is_dt_const_field(fld) = (
```

Comment:
you missed `@nospecialize` here

Reply:
Ah, yes. Last time I looked, I think we were missing that check in the caller, but looks like we now ensure it is an Int.